frontend/test/resolution/testCatch.cpp

```diff
@@ -0,0 +1,190 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "test-resolution.h"
+
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/resolution/resolution-queries.h"
+#include "chpl/types/CompositeType.h"
+
+// helper functions
+
+
+
+/* Check that, when resolving (or scope resolving) program,
+   identId within methodId resolves to refer to fieldId.
+   */
+static void testIt(const char* testName,
+                   const char* program,
+                   const char* errorTypeName,
+                   const char* catchIdStr,
+                   const char* argIdStr,
+                   const char* varIdStr) {
+  printf("test %s\n", testName);
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::get(context, testName);
+  std::string contents = program;
+  setFileText(context, path, contents);
+  // parse it so that Context knows about the IDs
+  const ModuleVec& vec = parseToplevel(context, path);
+
+  for (auto m : vec) {
+    m->dump();
+  }
+
+  // TODO: what else, if anything, to verify for the catch block's resolution?
+  ID catchId = ID::fromString(context, catchIdStr);
+  auto catchAst = parsing::idToAst(context, catchId);
+  assert(catchAst);
+  assert(catchAst->isCatch());
+
+  if (argIdStr) {
+    ID argId = ID::fromString(context, argIdStr);
+    auto argAst = parsing::idToAst(context, argId);
+    assert(argAst);
+    assert(argAst->isVariable());
+
+    auto argRes = resolvedExpressionForAst(context, argAst, nullptr, false);
+    auto argType = argRes->type().type();
+    assert(argType);
+    assert(argType->isClassType());
+    auto argName = argType->toClassType()->basicClassType()->name();
+    assert(argName == UniqueString::get(context, errorTypeName));
+  }
+
+  if (varIdStr) {
+    ID varId = ID::fromString(context, varIdStr);
+    auto varAst = parsing::idToAst(context, varId);
+    assert(varAst);
+    assert(varAst->isIdentifier());
+    auto varRes = resolvedExpressionForAst(context, varAst, nullptr, false);
+    auto varType = varRes->type().type();
+    assert(varType);
+    assert(varType->isClassType());
+    auto varName = varType->toClassType()->basicClassType()->name();
+    assert(varName == UniqueString::get(context, errorTypeName));
+  }
+}
+
+static void test1() {
+  testIt("test1.chpl",
+         R""""(
+            module M {
+              class SomeErrorName: Error { }
+              proc foo() throws { }
+              proc main() {
+                try {
+                  foo();
+                } catch e : SomeErrorName { }
+              }
+            }
+         )"""",
+         "SomeErrorName",
```

Comment:
For these tests, is it too much trouble to try and indicate where the ID is in the code? I can guess but I'm not certain. If it's too much of a PITA don't worry about it, though.